### PR TITLE
docs: explain use of `SecureStore` with Expo/React Native

### LINKED ISSUE #14523: Inconsistent store recommendations for React Native auth
# Improve documentation

## Link 
https://supabase.com/docs/guides/getting-started/tutorials/with-expo#initialize-a-react-native-app
https://supabase.com/docs/reference/javascript/initializing

## Describe the problem
In the starter guide for Expo, it was demonstrated using Expo SecureStore for the auth store, however, in the Javascript client docs, it said that it recommends using AsyncStorage as the store for the auth store.

Also, I started my project with the SecureStore setup and I encountered this warning, 
**Provided value to SecureStore is larger than 2048 bytes. An attempt to store such a value will throw an error in SDK 35.**
My related issue: https://github.com/orgs/supabase/discussions/14306

Not sure if the best way forward for Expo development is to use SecureStore, since there is a size limit.

## Describe the improvement
In the starter guide for Expo, replace the use of Expo's SecureStore with AsyncStorage to avoid confusion.

## Additional context
image
From the getting started with Expo tutorial
image
From the Supabase Javascript client docs

### PR DESCRIPTION
**Pull request by hf**
Provides an example of how [Expo's SecureStore](https://docs.expo.dev/versions/latest/sdk/securestore/) can be used with the Supabase Auth client.

Fixes:
- https://github.com/supabase/supabase/issues/14523

### CODE REVIEW

**Review by thorwebdev — APPROVED** [review]
LGTM, thanks 💚

File: apps/www/_blog/2023-08-01-react-native-storage.mdx [hunk 1]
```diff
@@ -86,15 +86,14 @@ npx create-expo-app@latest cloudApp --template tabs@49
 
 # Install dependencies
 npm i @supabase/supabase-js
-npm i react-native-url-polyfill base64-arraybuffer react-native-loading-spinner-overlay
+npm i react-native-url-polyfill base64-arraybuffer react-native-loading-spinner-overlay @react-native-async-storage/async-storage
 
 # Install Expo packages
-npx expo install expo-secure-store
```
Comment (MildTomato) [inline apps/www/_blog/2023-08-01-react-native-storage.mdx:92]: Some of the blog below still refers to `secure-store`

line 149

**Review by MildTomato — CHANGES_REQUESTED** [review]

File: apps/www/_blog/2023-08-01-react-native-storage.mdx (hunk 1 above)
Comment (MildTomato) [reply apps/www/_blog/2023-08-01-react-native-storage.mdx:92]: Screenshot 2023-09-26 at 3 32 33 PM

sorry, couldn't direct comment in PR. Should be able to find it.

Comment (thorwebdev) [reply apps/www/_blog/2023-08-01-react-native-storage.mdx:92]: Nice catch, thank you!

**Review by MildTomato — APPROVED** [review]

**Pull request merged**